src/gtc/common.py

```diff
@@ -304,16 +304,16 @@ class ScalarAccess(LocNode):
 class FieldAccess(LocNode):
     name: SymbolRef
     offset: CartesianOffset
-    data_index: List[int] = []
+    data_index: List[Any] = []
```

Comment:
Could this be `Union[int, Expr]` and `FieldAccess` becomes a GenericNode? Having `Any` in a strongly-typed IR seems like a workaround.

Reply:
That can be done